### PR TITLE
Use go 1.17 in our build environments

### PR DESCRIPTION
**Pull request by pintohutch**
* This fixes the `make presubmit` failures.
* Use Docker builds elsewhere by default.
* Build managed collection images as part of presubmit.
* Add documentation fixes.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -51,6 +51,84 @@ require (
 	sigs.k8s.io/yaml v1.2.0
 )
 
+require (
```
Comment (fabxc) [inline go.mod:54]: Why a separate block? 

Guess those are all indirect. But the above one also still contains indirect ones.

**Review by fabxc — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (pintohutch) [reply go.mod:44]: Not sure what happened there. Manually cleaned that up and reran `make presubmit`.

**Review by macxamin — APPROVED** [review]
Great Job! `make presubmit` works now.

**Comment by pintohutch** [comment]
Cool - I also spot checked this by deploying manually and doing various e2e tests in addition to the automated tests, as this involves bumping our build base images. Thanks for the reviews!

**Pull request merged**